src/NuGet.Core/NuGet.Build.Tasks/NuGet.targets

```diff
@@ -1364,6 +1369,63 @@ Copyright (c) .NET Foundation. All rights reserved.
     </PropertyGroup>
   </Target>
 
+  <!--
+    ============================================================
+    _GetRestoreTargetFrameworkOverride
+    ============================================================
+  -->
+  <Target Name="_GetRestoreTargetFrameworkOverride"
+          Condition=" '$(_DisableNuGetRestoreTargetFrameworksOverride)' != 'true' "
+          Returns="$(_TargetFrameworkOverride)">
+    <MSBuild
+        BuildInParallel="$(RestoreBuildInParallel)"
+        Condition=" '$(TargetFrameworks)' != '' "
+        Projects="$(MSBuildThisFileFullPath)"
+        Targets="_GetTargetFrameworksOverrides">
+
+      <Output
+          TaskParameter="TargetOutputs"
+          PropertyName="_TargetFrameworksOverride" />
+    </MSBuild>
+
+    <!-- Only set the override if TargetFrameworks has not been overriden as well. In that case, prefer it. -->
+    <MSBuild
+        BuildInParallel="$(RestoreBuildInParallel)"
+        Condition=" '$(TargetFrameworks)' != '' AND $(_TargetFrameworksOverride) == '' "
+        Projects="$(MSBuildThisFileFullPath)"
+        Targets="_GetTargetFrameworkOverrides">
```

Comment:
How much overhead is each `<MSBuild />` task call? Is there a way to run a single target and get both values at once?

Reply:
it's 1 ms according to CLP  for the `_GetRestoreTargetFrameworkOverride` target, but the individual target don't register. 

I think we should be able to use 1 target to get both. 

I'll try to fix it and run some tests again.

Reply:
Pushed a fix. 

Same results as before, but couldn't hurt to have 1 override instead of 2 :)